monai/transforms/spatial/functional.py

```diff
@@ -339,6 +363,80 @@ def resize(
     return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
 
 
+def _apply_affine_to_points(points, affine, include_shift: bool = True) -> torch.Tensor:
+    """
+    This internal function applies affine matrices to the point coordinate
+    Args:
+        points: point coordinates, Nx2 or Nx3 torch tensor or ndarray, representing [x, y] or [x, y, z]
+        affine: affine matrix to be applied to the point coordinates, sized (spatial_dims+1,spatial_dims+1)
+        include_shift: default True, whether the function apply translation (shift) in the affine transform
+    Returns:
+        transformed point coordinates, with same data type as ``points``, does not share memory with ``points``
+    """
+    # convert numpy to tensor if needed
+    points_t, *_ = convert_data_type(points, torch.Tensor)
+    points_t = points_t.to(dtype=COMPUTE_DTYPE)
+    affine_t, *_ = convert_to_dst_type(src=affine, dst=points_t)
+    spatial_dims = get_spatial_dims(points=points_t)
+
+    # compute new points
+    if include_shift:
+        # append 1 to form Nx(spatial_dims+1) vector, then transpose
+        points_affine = torch.cat(
+            [points_t, torch.ones(points_t.shape[0], 1, device=points_t.device, dtype=points_t.dtype)], dim=1
+        ).transpose(0, 1)
+        # apply affine
+        points_affine = torch.matmul(affine_t, points_affine)
+        # remove appended 1 and transpose back
+        points_affine = points_affine[:spatial_dims, :].transpose(0, 1)
+    else:
+        points_affine = points_t.transpose(0, 1)
+        points_affine = torch.matmul(affine_t[:spatial_dims, :spatial_dims], points_affine)
+        points_affine = points_affine.transpose(0, 1)
+
+    # convert tensor back to numpy if needed
+    points_affine, *_ = convert_to_dst_type(src=points_affine, dst=points)
+
+    return points_affine
+
+
+def resize_point(points, out_size, dtype, input_ndim, lazy, transform_info, **kwargs):
```

Comment:
How can we avoid having `*_point(...)` functions for each operation? At this level, shouldn't we be able to always just pass the correct transform? If we think about object space (raster space) and world space transforms then we shouldn't need to be performing these calculations based on spatial size here.

Reply:
As far as I understand it, resize should have no effect on point data, as it is entirely a "raster" space operation that preserves the extents of an image or volume, no?

Reply:
As discussed in the meeting, the alternative of taking the operation that defines the operation out of the function and putting it into its own implementation function works just as well as what I am suggesting.
```python
def resize_point(data, ...):
    meta_info = get_resize(...)
    out = apply_to_geom(data, metainfo)
    return out
```